routes/personal/personal.controller.js

```diff
@@ -79,3 +87,25 @@ const postMaritalStatus = (req, res) => {
 
   return res.redirect(req.body.redirect)
 }
+
+const postResidence = (req, res) => {
+  const errors = validationResult(req)
+
+  let residence = req.body.residence || null
+  req.session.personal = {
+    residence: residence,
+  }
+
+  if (!errors.isEmpty()) {
+    return res.status(422).render('personal/residence', {
+      data: { residence: req.body.residence } || {},
+      errors: errorArray2ErrorObject(errors),
+    })
+  }
+
+  if (errors.isEmpty() && req.body.residence === 'Yes') {
+    return res.redirect('address')
+  } else {
+    return res.redirect('/start')
+  }
```

Comment:
This is good, but just to be consistent with the style we've set, what do you think of:
```
if (req.body.residence !== 'Yes') {
 return res.redirect('/start')
}

return res.direct(req.body.redirect)
```

`req.bod.redirect` being the value we put in the hidden input on the page itself (mentioned in another comment)